dshr/dshr_mod.F90

```diff
@@ -1845,4 +1890,52 @@ subroutine dshr_pio_init(gcomp, sdat, logunit, rc)
 
   end subroutine dshr_pio_init
 
+  logical function dshr_check_restart_alarm(clock, rc)
```

Comment:
Could you please add a line or two of comments about our logic here. I think this would be helpful for future reference.